transaction-service/src/main/java/pl/adamboguszewski/transaction/service/application/TransactionService.java

```diff
@@ -25,26 +25,26 @@
     https://www.baeldung.com/spring-data-repositories
 */
 
-    private final TransactionRepository repository;
+    private final TransactionRepository<Transaction> repository;
 
-    public TransactionService(TransactionRepository repository) {
+    public TransactionService(TransactionRepository<Transaction> repository) {
         this.repository = repository;
     }
 
     public List<Transaction> getAllTransactions() {
-        return repository.findAll();
+        return repository.getAll();
```

Comment:
I just ctrl-c ctrl-v code from my different project, if you want you can change getAll to findAll etc again